src/AasxServerDB/ReadDB.cs

```diff
@@ -0,0 +1,169 @@
+﻿using AasCore.Aas3_0;
+using AdminShellNS;
+using Extensions;
+
+namespace AasxServerDB
+{
+    public class ReadDB
+    {
+        static public AdminShellPackageEnv GetPackageEnv(string path, AASSet aasDB) 
+        {
+            using (AasContext db = new AasContext())
+            {
+                if (path == null || path.Equals("") || aasDB == null)
+                    return null;
+
+                AssetAdministrationShell aas = new AssetAdministrationShell(
+                    id: aasDB.Identifier,
+                    idShort: aasDB.IdShort,
+                    assetInformation: new AssetInformation(AssetKind.Type, aasDB.GlobalAssetId),
+                    submodels: new List<AasCore.Aas3_0.IReference>());
+
+                AdminShellPackageEnv aasEnv = new AdminShellPackageEnv();
+                aasEnv.SetFilename(path);
+                aasEnv.AasEnv.AssetAdministrationShells.Add(aas);
+
+                var submodelDBList = db.SMSets
+                    .OrderBy(sm => sm.Id)
+                    .Where(sm => sm.AASId == aasDB.Id)
+                    .ToList();
+                foreach (var submodelDB in submodelDBList)
+                {
+                    Submodel sm = ReadDB.GetSubmodel(smDB:submodelDB);
+                    aas.Submodels.Add(sm.GetReference());
+                    aasEnv.AasEnv.Submodels.Add(sm);
+                }
+
+                return aasEnv;
+            }
+        }
+
+        static public Submodel GetSubmodel(SMSet? smDB = null, string? smIdentifier = "")

```

Comment:
```suggestion
        static public Submodel GetSubmodel(SMSet? smDB = null, string smIdentifier = "")

```
smIdentifier will never be null, but at least empty.